### PR TITLE
Enable windoze and macos in CI

### LINKED ISSUE #351: Enable windows and macos builds in CI

### PR DESCRIPTION
**Pull request by gadomski**
**Related Issue(s):** 

- Closes #351 

**Description:**

Once approved, we should enable these checks as required.

**PR Checklist:**

- [x] Code is formatted
- [x] Tests pass
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)
- [x] (once approved) Add windoze and macos to required checks

### CODE REVIEW

**Comment by pjhartzell** [comment]
>Once approved, we should enable these checks as required.

What does this mean?

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -33,8 +34,20 @@ jobs:
...
+        run: pip install .
+
+      - name: Install dev requirements
+        run: pip install -r requirements-dev.txt
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files
+
+      - name: Run pytest
+        run: pytest -Werror -s --block-network --cov pystac_client --cov-report term-missing
```
Comment (pjhartzell) [inline .github/workflows/continuous-integration.yml:47]: What is the `-Werror` option? Not finding that in the docs.

**Review by pjhartzell — APPROVED** [review]

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (gadomski) [reply .github/workflows/continuous-integration.yml:47]: Converts all warnings into errors (e.g. to catch places where we hit deprecations): https://docs.python.org/3/using/cmdline.html#cmdoption-W

File: pystac_client/stac_api_io.py [hunk 2]
```diff
@@ -288,3 +288,8 @@ def conforms_to(self, conformance_class: ConformanceClasses) -> bool:
     def set_conformance(self, conformance: Optional[List[str]]) -> None:
         """Sets (or clears) the conformance classes for this StacIO."""
         self._conformance = conformance
+
+
+def _is_url(href: str) -> bool:
```
[the remark this answers is not in the document]
Comment (gadomski) [reply pystac_client/stac_api_io.py:293]: Yeah, this beefed-up url detection was necessary to get CI passing on windoze: https://github.com/stac-utils/pystac-client/pull/378/commits/837792fd7c12bf1877e95b3dae2d8370ae1b55d8

**Comment by gadomski** [comment]
> What does this mean?

In the repository settings, you can add CI checks as required:

image

If a check isn't required, you can still merge the PR even if the check fails.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (pjhartzell) [reply .github/workflows/continuous-integration.yml:47]: Thanks.

**Pull request merged**